pkg/synchronizer/kubernetes/synchronizer.go

```diff
@@ -381,21 +381,17 @@ func isSpecialResource(gvr schema.GroupVersionResource) bool {
 func (sync *KubeSynchronizer) applyKindTemplates(res *ResourceMap, keySet map[string]bool) {
 	nri := sync.DynamicClient.Resource(res.GroupVersionResource)
 
-	for k, tplunit := range res.TemplateMap {
-		klog.V(1).Infof("k: %v, res.GroupVersionResource: %v", k, res.GroupVersionResource)
-
-		if !keySet[k] {
-			// ketSet contains resource keys to be deployed for this sinigle subscription item.
-			// if the current resource from the template map does not belong to ketSet,
-			// it belongs to another subscription. Skip.
-			klog.V(1).Infof("k: %v, does not belong to the order to be processed. skip", k)
-			continue
-		}
+	for resourceKey, okVal := range keySet {
+		tplunit := res.TemplateMap[resourceKey]
 
-		err := sync.applyTemplate(nri, res.Namespaced, k, tplunit, isSpecialResource(res.GroupVersionResource))
+		if okVal && tplunit != nil {
```

Comment:
If `tplunit` is nil I feel like we should log a message here. Otherwise I am good with the change. Thanks.